src/crypto/crypto_timing.cc

```diff
@@ -46,12 +47,31 @@ void TimingSafeEqual(const FunctionCallbackInfo<Value>& args) {
       CRYPTO_memcmp(buf1.data(), buf2.data(), buf1.size()) == 0);
 }
 
+bool FastTimingSafeEqual(Local<Value> receiver,
+                         const v8::FastApiTypedArray<uint8_t>& a,
+                         const v8::FastApiTypedArray<uint8_t>& b,
+                         // NOLINTNEXTLINE(runtime/references)
+                         FastApiCallbackOptions& options) {
+  uint8_t* data_a;
+  uint8_t* data_b;
+  if (a.length() != b.length() || !a.getStorageIfAligned(&data_a) ||
```

Comment:
Naively, if a.length() differs from b.length(), I would return false and not go any further. Is that considered timing unsafe?

Reply:
hmm. maybe this is already a security bug?!


In a REPL on node v21
```sh
> const { timingSafeEqual } = require('crypto')
> timingSafeEqual(Buffer.from('a'), Buffer.from('bc'))
Uncaught [RangeError: Input buffers must have the same byte length] {
  code: 'ERR_CRYPTO_TIMING_SAFE_EQUAL_LENGTH'
}
```

But if we return false in this fast method, will we throw also?

Reply:
I assume the main reason is that `node:crypto` mirrors lower-level cryptographic primitives, and timing-safe comparisons are usually implemented with a single length parameter only. This forces the caller to consider whether disclosing differences in length might be a side channel.

In any case, the fast API function should behave like the regular implementation, so if anyone wants to change this behavior, that'd have to be a separate discussion/change.

> But if we return false in this fast method, will we throw also?

Note that I am setting `options.fallback = true` before returning, which causes V8 to call the regular "slow" function.